packages/common/src/common.ts

```diff
@@ -384,7 +384,7 @@ export class Common {
       if (mergeIndex >= 0 && td !== undefined && td !== null) {
         if (hfIndex >= mergeIndex && BigInt(hfs[mergeIndex].ttd!) > td) {
           throw Error('Maximum HF determined by total difficulty is lower than the block number HF')
-        } else if (hfIndex < mergeIndex && BigInt(hfs[mergeIndex].ttd!) <= td) {
+        } else if (hfIndex < mergeIndex && BigInt(hfs[mergeIndex].ttd!) < td) {
```

Comment:
Ok, this change triggers a Common test. I am pretty sure this is ok, we should check for `<` not `<=`?

Reply:
Ah and I forgot to note: this change was triggered by a hive test with thus this exact scenario of hitting the TTD exactly.